packages/main/src/RPA/Robocorp/WorkItems.py

```diff
@@ -948,50 +960,54 @@ def _load_adapter(self, default) -> Type[BaseAdapter]:
 
     @staticmethod
     def _interpret_content(body: str) -> Union[dict, str]:
-        loaders = [json.loads, yaml.full_load]
-        for loader in loaders:
+        loaders = {"JSON": json.loads, "YAML": yaml.full_load}
+        for name, loader in loaders.items():
             try:
                 body = loader(body)
             except Exception as exc:  # pylint: disable=broad-except
                 logging.debug(
                     "Failed deserializing input e-mail body content with loader %r "
                     "due to: %s",
-                    loader.__name__,
+                    name,
                     exc,
                 )
             else:
                 break
 
         return body
 
+    def _get_email_content(
+        self, variables: Dict
+    ) -> Optional[Tuple[str, bool, Tuple[str]]]:
+        # Returns the extracted e-mail [parsed] content and its payload destination.
+        to_tuple = (
+            lambda keys: keys if isinstance(keys, tuple) else (keys,)
+        )  # noqa: E731
+        file_list = self.list_work_item_files()
+        for input_keys, output_keys in self._auto_parse_email.items():
+            input_keys = to_tuple(input_keys)
+            for input_key in input_keys:
+                content = get_dot_value(variables, input_key)
+                if not content and input_key in file_list:
+                    path = Path(self.get_work_item_file(input_key))
+                    content = path.read_text(encoding=ENCODING)
+                    path.unlink()
```

Comment:
Question... Why do we unlink the file after reading?
Security reasons? Cleaning up?

Reply:
Clean-up because is an undesired behaviour of the Work Item file downloaded content in order to get the data itself. An alternative would be to download it in the output dir (git-ignored), but still we have to remove it since the user didn't ask for it. (we pull it to get the e-mail content)